operators/src/processing/time_projection/mod.rs

```diff
@@ -0,0 +1,560 @@
+use std::sync::Arc;
+
+use crate::engine::{
+    ExecutionContext, InitializedVectorOperator, Operator, QueryContext, SingleVectorSource,
+    TypedVectorQueryProcessor, VectorOperator, VectorQueryProcessor, VectorResultDescriptor,
+};
+use crate::util::Result;
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+use geoengine_datatypes::collections::{
+    FeatureCollection, FeatureCollectionInfos, FeatureCollectionModifications,
+};
+use geoengine_datatypes::primitives::{Geometry, TimeInterval};
+use geoengine_datatypes::primitives::{TimeInstance, TimeStep, VectorQueryRectangle};
+use geoengine_datatypes::util::arrow::ArrowTyped;
+use log::debug;
+use rayon::iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator};
+use rayon::ThreadPool;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt, Snafu};
+
+/// Projection of time information in queries and data
+///
+/// This operator changes the temporal validity of the queried data.
+/// In order to query all valid data, it is necessary to change the query rectangle as well.
+///
+// TODO: extend for rasters
+pub type TimeProjection = Operator<TimeProjectionParams, SingleVectorSource>;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct TimeProjectionParams {
+    /// Specify the time step granularity and size
+    step: TimeStep,
+    /// Define an anchor point for `step`
+    step_reference: Option<TimeInstance>,
+}
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)), context(suffix(false)), module(error))]
+pub enum TimeProjectionError {
+    #[snafu(display("Time step must be larger than zero"))]
+    WindowSizeMustNotBeZero,
+
+    #[snafu(display("Query rectangle expansion failed: {}", source))]
+    CannotExpandQueryRectangle {
+        source: geoengine_datatypes::error::Error,
+    },
+
+    #[snafu(display("Feature time interval expansion failed: {}", source))]
+    CannotExpandFeatureTimeInterval {
+        source: geoengine_datatypes::error::Error,
+    },
+}
+
+#[typetag::serde]
+#[async_trait]
+impl VectorOperator for TimeProjection {
+    async fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedVectorOperator>> {
+        ensure!(self.params.step.step > 0, error::WindowSizeMustNotBeZero);
+
+        let source = self.sources.vector.initialize(context).await?;
+
+        debug!(
+            "Initializing `TemporalRasterAggregation` with {:?}.",
+            &self.params
+        );
+
+        let step_reference = self
+            .params
+            .step_reference
+            // use UTC 0 as default
+            .unwrap_or_else(|| TimeInstance::from_millis_unchecked(0));
```

Comment:
this needs to be very prominent in the docs i guess.

Reply:
[20669f7](https://github.com/geo-engine/geoengine/pull/478/commits/20669f7e60b0ab37c250fdf3b1daaa807622f434)